library/std/src/path.rs

```diff
@@ -2706,15 +2715,17 @@ impl cmp::Eq for Path {}
 impl cmp::PartialOrd for Path {
     #[inline]
     fn partial_cmp(&self, other: &Path) -> Option<cmp::Ordering> {
-        self.components().partial_cmp(other.components())
+        Some(Ord::cmp(self, other))
     }
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
 impl cmp::Ord for Path {
     #[inline]
     fn cmp(&self, other: &Path) -> cmp::Ordering {
-        self.components().cmp(other.components())
+        self.components()
+            .cmp(other.components())
```

Comment:
It might be better to use `Ord::cmp` here too, while implementing #86898 I ran into some ambiguities around `Ord::cmp` vs. `Iterator::cmp`.